src/main/java/ch/njol/skript/expressions/ExprEntities.java

```diff
@@ -129,62 +143,81 @@ public boolean isLoopOf(String s) {
+		if (isUsingRadius || isUsingCuboid) {
+			Iterator<? extends Entity> iter = iterator(event);
 			if (iter == null || !iter.hasNext())
 				return null;
 
-			List<Entity> l = new ArrayList<>();
+			List<Entity> list = new ArrayList<>();
 			while (iter.hasNext())
-				l.add(iter.next());
-			return l.toArray((Entity[]) Array.newInstance(returnType, l.size()));
+				list.add(iter.next());
+			return list.toArray((Entity[]) Array.newInstance(returnType, list.size()));
 		} else {
 			if (chunks != null) {
-				return EntityData.getAll(types.getArray(e), returnType, chunks.getArray(e));
+				return EntityData.getAll(types.getArray(event), returnType, chunks.getArray(event));
 			} else {
-				return EntityData.getAll(types.getAll(e), returnType, worlds != null ? worlds.getArray(e) : null);
+				return EntityData.getAll(types.getAll(event), returnType, worlds != null ? worlds.getArray(event) : null);
 			}
 		}
 	}
 
 	@Override
 	@Nullable
 	@SuppressWarnings("null")
-	public Iterator<? extends Entity> iterator(Event e) {
+	public Iterator<? extends Entity> iterator(Event event) {
 		if (isUsingRadius) {
 			assert center != null;
-			Location l = center.getSingle(e);
-			if (l == null)
+			Location location = center.getSingle(event);
+			if (location == null)
 				return null;
 			assert radius != null;
-			Number n = radius.getSingle(e);
-			if (n == null)
+			Number number = radius.getSingle(event);
+			if (number == null)
 				return null;
-			double d = n.doubleValue();
+			double rad = number.doubleValue();
 
-			if (l.getWorld() == null) // safety
+			if (location.getWorld() == null) // safety
 				return null;
 
-			Collection<Entity> es = l.getWorld().getNearbyEntities(l, d, d, d);
-			double radiusSquared = d * d * Skript.EPSILON_MULT;
-			EntityData<?>[] ts = types.getAll(e);
-			return new CheckedIterator<>(es.iterator(), e1 -> {
-					if (e1 == null || e1.getLocation().distanceSquared(l) > radiusSquared)
+			Collection<Entity> nearbyEntities = location.getWorld().getNearbyEntities(location, rad, rad, rad);
+			double radiusSquared = rad * rad * Skript.EPSILON_MULT;
+			EntityData<?>[] entityTypes = types.getAll(event);
+			return new CheckedIterator<>(nearbyEntities.iterator(), entity -> {
+					if (entity == null || entity.getLocation().distanceSquared(location) > radiusSquared)
 						return false;
-					for (EntityData<?> t : ts) {
-						if (t.isInstance(e1))
+					for (EntityData<?> entityType : entityTypes) {
+						if (entityType.isInstance(entity))
 							return true;
 					}
 					return false;
 				});
+		} else if (isUsingCuboid) {
+			assert from != null;
+			Location corner1 = from.getSingle(event);
+			if (corner1 == null)
+				return null;
+			assert to != null;
+			Location corner2 = to.getSingle(event);
+			if (corner2 == null)
+				return null;
+			EntityData<?>[] entityTypes = types.getAll(event);
+			Collection<Entity> entities = corner1.getWorld().getNearbyEntities(BoundingBox.of(corner1, corner2));
```

Comment:
Ah, please be careful here because locations don't always have worlds attached (usually because users made them without one, they got the location from an addon/plugin, or there was some vector conversion sadness idk)
so this could throw an NPE since you're just assuming corner 1 has a world.

You could look for a world from _either_ corner 1 _or_ 2. If neither have one I think it's okay to say nothing's in the box, but you could do what we did with teleports and look for a contextual location. Not sure how to do that here though.

Reply:
Ill definitely check both corners but if neither have a world would it make sense to try to get the default world, or maybe the first world in the world list? Idk if that would be better than just returning 0 though. I would want to add some kind of warning but I don't think its possible to know at parse time

Reply:
Just return null if the worlds are both null. You won't be able to get the location during parsing and if there is no world at runtime, you won't be able to gather the entities.